File: backend/app/scenarios.py
from typing import Dict, Any

SCENARIOS: Dict[str, Dict[str, Any]] = {
    "casual": {
        "id": "casual",
        "title": "Casual Talk",
        "category": "Daily Life",
        "icon": "☕️",
        "level": "All Levels",
        "description": "Have a relaxed, fun conversation about life, music, hobbies, and random hot takes.",
        "greeting": "Yo! Alex here. If you could teleport anywhere on Earth right this second, where are we dropping in?",
        "system_prompt": """You are Alex, an unconventional, witty, and playfully curious American on a live voice phone call with the user.

YOUR PERSONALITY:
- NEVER follow predictable, boring AI templates.
- Drop playful hot-takes, funny dilemmas, quick banter, and unexpected angles.
- Keep responses ULTRA-PUNCHY: exactly 1 to 2 short sentences (under 18-20 words total).
- Always output ONLY your spoken words starting directly with "Alex: ".
"""
    },
    "airport": {
        "id": "airport",
        "title": "Airport & Customs",
        "category": "Travel",
        "icon": "✈️",
        "level": "Intermediate",
        "description": "Pass through US passport control and customs at JFK Airport in New York.",
        "greeting": "Next in line, please! Step forward. Welcome to New York. May I see your passport and declaration form?",
        "system_prompt": """You are Officer Alex, a professional, firm, yet polite US Customs and Border Protection officer at JFK Airport.

YOUR ROLE:
- Interview the traveler entering the United States.
- Ask realistic border questions one at a time: purpose of visit, length of stay, where they are staying, if they have return tickets, items in luggage.
- React realistically to their answers (e.g. asking for hotel name or business details if needed).
- Keep responses ULTRA-PUNCHY: exactly 1 to 2 short sentences (under 18 words).
- Always output ONLY your spoken words starting directly with "Alex: ".
"""
    },
    "job_interview": {
        "id": "job_interview",
        "title": "Job Interview",
        "category": "Career",
        "icon": "💼",
        "level": "Advanced",
        "description": "Practice a tech / product job interview with a sharp international recruiter.",
        "greeting": "Hi there! Thanks for taking the time today. Have a seat! To kick things off, tell me a little about yourself.",
        "system_prompt": """You are Alex, an experienced and engaging Lead Recruiter conducting a live job interview.

YOUR ROLE:
- Ask real, sharp interview questions one by one (background, technical challenges, working under pressure, conflict resolution).
- React briefly to their response before asking the next follow-up.
- Keep responses ULTRA-PUNCHY: exactly 1 to 2 concise sentences (under 20 words).
- Always output ONLY your spoken words starting directly with "Alex: ".
"""
    },
    "restaurant": {
        "id": "restaurant",
        "title": "Cafe & Restaurant",
        "category": "Food & Dining",
        "icon": "🍕",
        "level": "Beginner / All",
        "description": "Order delicious food, drinks, and banter with a lively Brooklyn waiter.",
        "greeting": "Hey folks, welcome to Joe's Bistro! My name is Alex. Can I get you started with something cold to drink?",
        "system_prompt": """You are Alex, an energetic, friendly waiter at a bustling Brooklyn bistro.

YOUR ROLE:
- Guide the guest through ordering drinks, appetizers, and main courses.
- Suggest daily specials (e.g. truffle pasta, spicy pepperoni), handle modifications (extra sauce, allergies), and bring the check.
- Keep the vibe upbeat, warm, and natural.
- Keep responses ULTRA-PUNCHY: exactly 1 to 2 short sentences (under 18 words).
- Always output ONLY your spoken words starting directly with "Alex: ".
"""
    }
}

DEFAULT_SCENARIO_ID = "casual"

def get_scenario(scenario_id: str) -> Dict[str, Any]:
    return SCENARIOS.get(scenario_id, SCENARIOS[DEFAULT_SCENARIO_ID])

File: frontend/src/data/scenarios.ts
import type { Scenario } from '../types';

export const SCENARIOS: Scenario[] = [
  {
    id: 'casual',
    title: 'Casual Talk',
    category: 'Daily Life',
    icon: '☕️',
    level: 'All Levels',
    description: 'Легкий, остроумный разговор о жизни, хобби и планах.',
    color: 'from-amber-500/20 to-orange-500/20 border-amber-500/40 text-amber-300'
  },
  {
    id: 'airport',
    title: 'Airport & Customs',
    category: 'Travel',
    icon: '✈️',
    level: 'Intermediate',
    description: 'Паспортный контроль и таможня в аэропорту JFK Нью-Йорка.',
    color: 'from-sky-500/20 to-blue-500/20 border-sky-500/40 text-sky-300'
  },
  {
    id: 'job_interview',
    title: 'Job Interview',
    category: 'Career',
    icon: '💼',
    level: 'Advanced',
    description: 'Реалистичное собеседование с международным tech-рекрутером.',
    color: 'from-purple-500/20 to-indigo-500/20 border-purple-500/40 text-purple-300'
  },
  {
    id: 'restaurant',
    title: 'Cafe & Dining',
    category: 'Food',
    icon: '🍕',
    level: 'Beginner / All',
    description: 'Заказ еды, напитков и живой диалог с официантом в кафе.',
    color: 'from-emerald-500/20 to-teal-500/20 border-emerald-500/40 text-emerald-300'
  }
];
